lib/beautiful/xresources.lua

```diff
@@ -101,13 +101,15 @@ end
 
 --- Set DPI for a given screen (defaults to global).
 -- @tparam number dpi DPI value.
--- @tparam[opt] integer s Screen.
+-- @screen[opt] s The screen.
 function xresources.set_dpi(dpi, s)
-    s = get_screen(s)
     if not s then
         xresources.dpi = dpi
     else
-        dpi_per_screen[s] = dpi
+        s = get_screen(s)
+        if s then
+            dpi_per_screen[s] = dpi
+        end
```

Comment:
Documentation-wise: Sure, thanks for catching this.
However, the code-change shouldn't make any real difference, should it? (Well, you can no longer use e.g. "primary" or "HDMI-1" as arguments, but that's fine with me)
Only thing I see is that using e.g. a function as argument no longer throws an error.

What kind of invalid arguments were you thinking of here? Should that be mentioned in the commit message?